packages/connectivity/connectivity_platform_interface/lib/connectivity_platform_interface.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'method_channel_connectivity.dart';
+
+/// The interface that implementations of connectivity must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `Connectivity`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [ConnectivityPlatform] methods.
+abstract class ConnectivityPlatform extends PlatformInterface {
+  /// Constructs a ConnectivityPlatform.
+  ConnectivityPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static ConnectivityPlatform _instance = MethodChannelConnectivity();
+
+  /// The default instance of [ConnectivityPlatform] to use.
+  ///
+  /// Defaults to [MethodChannelConnectivity].
+  static ConnectivityPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [ConnectivityPlatform] when they register themselves.
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(ConnectivityPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Checks the connection status of the device.
+  Future<String> checkConnectivity() {
```

Comment:
Can this return an `Enum` of standardized connectivity statuses? Using a `String` makes it harder to know what values the core plugin expects.

Reply:
(Apologies for the late review, I've just noticed this was a thing because of pub.dev notifications) /cc @amirh

Reply:
It will, once I hook it up to `connectivity.dart` in the next PR. But as for the method_channel, it does return a String from the native platforms.

Reply:
Well, it doesn't matter what it returns from the method_channel, because you have an implementation that can convert whatever is returned from the native side into the standardized Enum.

If you're planning on eventually returning an Enum from the user-facing plugin, that Enum can be part of the platform interface package as well.